### DISCUSSION TITLE
snippet

### DISCUSSION THREAD

**Discussion by f14m07** [liedekef/events-made-easy#67] [General]
Hi,
I have been using this snippet adapted from an example in the EME documentation for a long time. This morning it did not work. I checked the mailbox and everything works. Maybe there was a name change in the EME code?
Here is the snippet
```add_action('eme_insert_event_action','eme_mail_event');
add_action('emefs_submit_event_action','eme_mail_event');
function eme_mail_event ($event) {
   $contact = eme_get_event_contact ($event);
   $contact_email = $contact->user_email;
   $contact_name = $contact->display_name;
   $subject_format="Vogue et Rêve - Une nouvelle navigation a été programmée";
   $body_format="Bonjour <br>Un nouvel évènement a été mis en ligne sur Vogue et Rêve. <br> 
   Son nom est <b> #_EVENTNAME </b>, <br> 
   Il a lieu le <b>#_STARTDATE à #_STARTTIME</b> organisé par #_AUTHORDISPNAME <br><br>
   #_LOCATIONIMAGETHUMB<br>
   Vous serez sur le voilier <b>#_LOCATIONNAME</b> 
   <br>
   <br>
   Vous devrez vous identifier pour réserver <a href=#_EVENTPAGEURL ?lang=fr>cliquez ici</a> <br>
   <br>
   Si vous souhaitez vous désinscrire de ce type de message <a href=#_UNSUB_URL>cliquez ici</a> <br><br>
   ";
   $subject=eme_replace_placeholders($subject_format, $event, "text");
   $body=eme_replace_placeholders($body_format, $event);
   $person_ids=eme_get_groups_person_ids(6);
    //si evenement pas un brouillon envoyer le mailing
    if ($event['event_status'] != EME_EVENT_STATUS_DRAFT) {
   foreach ( $person_ids as $person_id ) {
      $person=eme_get_person($person_id);
      $person_name=eme_format_full_name($person['firstname'],$person['lastname']);
      eme_send_mail($subject,$body, $person['email'], $person_name, $contact_email, $contact_name);
   }
}
}```

**Comment by f14m07** [comment]
You have made changes to the groups recently. Could that be where it's coming from?

**Comment by liedekef** [comment]
I don't see a real issue in the code (next time use the 3 backticks to include your code, it will render it better then).
The only thing I can think about: is your contact mail an email that is allowed to send emails? I did changes there, so maybe that is interfering ... 
You can try to add some prints in the eme_send_mail function, to see if the mail goes through. Or if using smtp, activate smtp debugging. And maybe check your mailserver logs?

**Comment by f14m07** [comment]
I tried with 'Event related email' sent to a 'group' and the email did not go away (sending the preview email works). 
When testing with a generic email sent to a 'group' the email did not go away (sending the preview email works).
I then sent a generic email to a number of people and it worked very well.
So it seems that the problem is probably in the selection of the recipients of the 'group'.

**Comment by liedekef** [comment]
That is not related to the initial problem, although that might be an issue. I did change sql statements (to remove the "AS groups" part because newer mysql consider "groups" as a reserved word). I'll check that out, but your initial problem is not related here

**Comment by f14m07** [comment]
I discovered the issue because the snippet stopped to works. As this snippet uses `$person_ids=eme_get_groups_person_ids(6);` I thought it could be a groups problem. After that, to check whether the groups or the event are the problem I sent mail with 'event related email'. And only after I did the test with generic email.

**Comment by liedekef** [comment]
Indeed, I needed to look again at the snippet to see that call :-)
Can you check if this helps:
https://github.com/liedekef/events-made-easy/commit/3c3c123fe46a9d360bbd465499941f97e0c6b265

**Reply by liedekef** [reply]
It was confirmed this fixes it. I'll release a new version.
